### DISCUSSION TITLE
SKTP

### DISCUSSION THREAD

**Discussion by slesinger** [hpingel/Sidekick64#3] [Q&A]
Hello, SKTP looks very appealing. 
1. Is the protocol documentation available somewhere?
2. Is sktp.php demo source code available as example?
3. Is SKTP browser (C64 application) source available?
Gregory Nacu writes about [mtext as a c64 convenient format](https://c64os.com/post/conversionservices) that can be proxied from html pages of normal Internet. Sidekick and SKTP can be the best and fastest protocol for such a proxy. I wondar should connecting these dots bring something useful. Thank you.

PS: Thanks for your great contribuiton to SK64. It makes a big thing out of it.

**Comment by hpingel** [comment]
Hello @slesinger,

Thank you for your questions, I just realized that I missed your posting here for quite a while. Sorry for the late answer - but now in 2024 I may have better answers:

1. Protocol documentation is implicit in the example code in https://github.com/hpingel/sktp-server . I can write a documentation but so far the interest was limited.
2. See the link in (1). I released the stuff in 2023 (not all SKTP apps are released at the moment, but the basis and examples).
3. The SKTP browser (or SKTP client) "on" C64 is able to run with Sidekick64 and since 2023 also with WiC64. The source code for Sidekick64 is in C++ in this repo right here. The assembly code for the WiC64 is available here since 2023: https://github.com/hpingel/sktp-client. I haven't read about mtext, just had a quick glance. Not sure if I need mtext to do what I want to do.

**Reply by amiga23** [reply]
I am also interested. Currently I am rewriting your example in python, just for learning.
Would like to implement something, that can provide content with your SKTP Protocol, QuantumLink and simple telnet. Maybe add some home-automation functionality.

**Reply by hpingel** [reply]
@amiga23 : Could you open an issue on Project SKTP-Server and point me to your code and I will provide you the details you need.
